source/extensions/filters/network/common/redis/redis_command_stats.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/stats/scope.h"
+#include "envoy/stats/timespan.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+#include "extensions/filters/network/common/redis/codec.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class RedisCommandStats {
+public:
+  RedisCommandStats(Stats::Scope& scope, const std::string& prefix, bool enabled);
+
+  Stats::Counter& counter(std::string name);
```

Comment:
per earlier comments I'm hoping we can remove the string-passing form of these.

But if you *did* need to pass strings you should do so by const ref.

Reply:
Joshua, I'm afraid there isn't a good way to avoid the string-passing.

In `client_impl.cc`, we have a request which contains an arbitrary redis command, represented as a string inside a `RespValue`. We will always need some way to map this object, or rather the command string part of it, to a specific `StatName`:
```
PoolRequest* ClientImpl::makeRequest(const RespValue& request, PoolCallbacks& callbacks) {
  ASSERT(connection_->state() == Network::Connection::State::Open);

  const bool empty_buffer = encoder_buffer_.length() == 0;

  std::string to_lower_command(redis_command_stats_->getCommandFromRequest(request));
  to_lower_table_.toLowerCase(to_lower_command);
  pending_requests_.emplace_back(*this, callbacks, to_lower_command);
```

Reply:
I think I addressed this above. Sorry I am working entirely on my phone now and it is hard to keep track :)

Reply:
I think I addressed this above. Sorry I am working entirely on my phone now and it is hard to keep track :)